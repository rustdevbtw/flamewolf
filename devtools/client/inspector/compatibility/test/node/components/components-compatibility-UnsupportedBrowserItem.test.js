/* Any copyright is dedicated to the Public Domain.
   http://creativecommons.org/publicdomain/zero/1.0/ */

"use strict";

/**
 * Unit tests for the UnsupportedBrowserItem component.
 */

const { shallow } = require("enzyme");
const React = require("react");

const BrowserItem = React.createFactory(
  require("resource://devtools/client/inspector/compatibility/components/UnsupportedBrowserItem.js")
);

describe("UnsupportedBrowserItem component", () => {
  it("renders the browser", () => {
    const item = shallow(
      BrowserItem({
        id: "firefox",
        name: "Firefox",
        version: "113",
        unsupportedVersions: [
          {
            status: "current",
            version: "113",
          },
          {
            status: "beta",
            version: "114",
          },
          {
            status: "release",
            version: "115",
          },
        ],
      })
    );
    expect(item).toMatchSnapshot();
  });
});
