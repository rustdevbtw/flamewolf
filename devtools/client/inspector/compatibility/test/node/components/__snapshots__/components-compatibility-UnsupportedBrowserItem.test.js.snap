// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnsupportedBrowserItem component renders the browser 1`] = `
<Localized
  $browsers="Firefox 113 (current)
Firefox 114 (beta)
Firefox 115 (release)"
  attrs={
    Object {
      "title": true,
    }
  }
  id="compatibility-issue-browsers-list"
>
  <li
    className="compatibility-browser"
    data-browser-id="firefox"
  >
    <BrowserIcon
      id="firefox"
      name="Firefox"
    />
    <span
      className="compatibility-browser-version"
    >
      113
    </span>
  </li>
</Localized>
`;
